### PR TITLE
Slurm

### PR DESCRIPTION
**Pull request by Dawid64**

### CODE REVIEW

**Comment by Tole-k** [comment]
important note, OneHotEncoder does nothing to target feature at the moment, it just returns it unchanged because it doesn't really make sense to transform it this way, moreover it would convert target from 1D pd.Series into 2D pd.Dataframe. To process target feature we have LabelEncoder which mitigates this problem while still encoding strings into numeric values

**Comment by Tole-k** [comment]
I believe we wanted to remove CatBoost from our project over ties with Russia, I actually have already done it but on the "algorithms" branch which never got merged

**Comment by Tole-k** [comment]
It would be nice to have an option to extract both Statistical and Topological features but get_sota_meta_features returns one or the other

File: turbo_ml/meta_learning/dataset_parameters/sota.py
Comment (Tole-k) [inline turbo_ml/meta_learning/dataset_parameters/sota.py:1]: It would be nice to have an option to extract both Statistical and Topological features but get_sota_meta_features returns one or the other

File: turbo_ml/workflow/extract_parameters.py
Comment (Tole-k) [inline turbo_ml/workflow/extract_parameters.py:1]: important note, OneHotEncoder does nothing to target feature at the moment, it just returns it unchanged because it doesn't really make sense to transform it this way, moreover it would convert target from 1D pd.Series into 2D pd.Dataframe. To process target feature we have LabelEncoder which mitigates this problem while still encoding strings into numeric values

File: turbo_ml/algorithms/cat_boost.py
Comment (Tole-k) [inline turbo_ml/algorithms/cat_boost.py:1]: I believe we wanted to remove CatBoost from our project over ties with Russia, I actually have already done it but on the "algorithms" branch which never got merged

File: turbo_ml/workflow/algorithms_evaluations.py
Comment (Tole-k) [inline turbo_ml/workflow/algorithms_evaluations.py:1]: Why is slice index being mutliplied by 10 here?

**Review by Tole-k — CHANGES_REQUESTED** [review]

File: turbo_ml/workflow/extract_parameters.py
Comment (Dawid64) [reply turbo_ml/workflow/extract_parameters.py:1]: Can you combine OHE and label encoder into one? such that we can pass encoders as one list for both data and target, we can alternately make an argument for OHE / Label if should it change target or data, unless you have some other propositions

Comment (Tole-k) [reply turbo_ml/workflow/extract_parameters.py:1]: Yes, i have added label_encoder functionality to one_how_encoder, and renamed it to just 'encoder' not to mislead

File: requirements.txt [hunk 1]
```diff
@@ -1,112 +1,21 @@
...
-virtualenv==20.26.3
-wcwidth==0.2.13
-xattr==1.1.0
-xgboost==2.1.1
-zipp==3.20.0
+# Base
+pytest
+pandas
+numpy
+scikit-learn
```
Comment (adamtms) [inline requirements.txt:5]: During prediction we have static amount of scikit-learn models defined in the file but during training it is chosen automatically. Maybe we should choose specific scikit-learn version so the number of models would not change or have the available models available for predicition also chosen dynamically not hard-coded.

Comment (Dawid64) [reply requirements.txt:5]: Good idea the best idea is probably saving models into pickle together with preprocessing and config

File: turbo_ml/meta_learning/meta_model_search.py
Comment (Tole-k) [inline turbo_ml/meta_learning/meta_model_search.py:1]: I am wondering shouldn't the order of __MODELS_NAMES__ match the order of columns in algorithm_results.csv ?, because i don't think it does

File: turbo_ml/meta_learning/meta_model_search.py [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+from turbo_ml.base.model import Model
+from turbo_ml.utils import options
+from .model_prediction.model_prediction import Predictor
+from turbo_ml.base import get_models_list
+import pandas as pd
+import torch
+from .model_architecture import ModelArchitecture
+import os
+
+__MODELS_NAMES__ = ["NeuralNetworkModel", "XGBoostClassifier", "AdaBoostClassifier", "BaggingClassifier", "BernoulliNB", "CalibratedClassifierCV", "CategoricalNB", "ComplementNB", "DecisionTreeClassifier", "DummyClassifier", "ExtraTreeClassifier", "ExtraTreesClassifier", "GaussianNB", "GaussianProcessClassifier", "GradientBoostingClassifier", "HistGradientBoostingClassifier", "KNeighborsClassifier",
```
Comment (Tole-k) [inline turbo_ml/meta_learning/meta_model_search.py:13]: I am wondering shouldn't the order of MODELS_NAMES match the order of columns in algorithm_results.csv ?, because i don't think it does

What's worse is that i believe it uses this __MODELS_NAMES__ while predicting even when the model was trained on algorithm_families and not our sklearn algorithms, so it takes an index of best performing families and returns __MODEL_NAME__ associated with it which results i total gibberish being returned

ehh, i think we need to deliniate stronger between our system behavoiur when working with family scores and algorithm scores
